boltdesignsystem/bolt
# DS-409 Update plop component generator

## Jira

https://pegadigitalit.atlassian.net/browse/DS-409

## Summary

Schema, twig and test templates were updated to our standards

## Details

A branch [feature/DS-408-Convert-Yeoman-to-Plop-alternative](https://github.com/boltdesignsystem/bolt/pull/2274) was pulled to this one.

- A new prompt was added to the component generator that will render a different JS, Jest Test and Twig file if this component is a web component.
- `title` and `content` were added to `component.schema`
- `component.docs.twig` was updated with single quotes
-  `component.twig` was updated with single quotes, `attributes` was changed with `props.class`, `title` and `content` were added to the template
- `component.test.js` was updated with commented code in case to use, according to the master schema
- `element.docs.twig` was updated with single quotes

## How to test

1. Pull the branch locally and run `yarn setup`.
2. Run `yarn cc` to generate component and check if all the files are generated correctly (remember to run `yarn` after).
3. On the "Is this component a web component?:" press enter ("n" is the default value)
4. Run `yarn cc` to generate component and check if all the files are generated correctly (remember to run `yarn` after).
5. On the "Is this component a web component?:" question type "y"
6. Run `yarn ce` to generate element and check if all the files are generated correctly (remember to run `yarn` after).
